### PR TITLE
Dialtone V6 Component Table Implementation

### LINKED ISSUE #251: [CSS] Component: Tables
**Acceptance Criteria**

- Update the Table component Dialtone classes

**Resources**

Design: https://www.figma.com/file/zP2uyVxAg4notGNZwW64mW/Components-(Dialtone-6.0)?node-id=5544%3A21729

### PR DESCRIPTION
**Pull request by ryandialpad**
# Dialtone V6 Component Table Implementation

## :hammer_and_wrench: Type Of Change

- [ ] Fix
- [x] Feature
- [x] Refactoring
- [ ] Documentation

## :book: Description

Updated Table css component
Updated Table documentation

## :bulb: Context

Closes #251 

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [x] Update, remove, or extend all affected documentation.
 - [x] Provide a description what is being changed, extended, or removed.
 - [x] Link to any issue(s) this request is resolving.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
 - [x] Request a review from Drew Chandler, Joshua Hynes, or Ted Goas.

## :camera: Screenshots / GIFs

Screen Shot 2021-04-05 at 3 01 38 PM

### CODE REVIEW

File: lib/less/components/tables.less [hunk 1]
```diff
@@ -55,24 +76,7 @@
...
-    }
-}
-
-//  ============================================================================
-//  $   STRIPED ROW BACKGROUNDS
-//  ----------------------------------------------------------------------------
-.d-table--striped {
-    tr:nth-child(even) {
-        background-color: hsl(var(--black-100-h) var(--black-100-s) calc(var(--black-100-l) + 2.5%));
-    }
```
Comment (ryandialpad) [inline lib/less/components/tables.less]: Removed unused variant

File: lib/less/components/tables.less [hunk 2]
```diff
@@ -13,40 +13,61 @@
...
-    tbody tr:last-of-type {
+    // Nested Table Body, First Row Styles - TODO: is this still required?
+    tbody + tbody tr:first-of-type {
+        // Cell Styles
         td,
         th {
-            border-bottom-width: 0;
+            border-top: var(--su2) solid var(--table--bc);
         }
     }
```
Comment (ryandialpad) [inline lib/less/components/tables.less]: @dialpad-drew  Do we need this style anymore? I didn't see it in the design spec

File: lib/less/components/tables.less [hunk 3]
```diff
@@ -13,40 +13,61 @@
 //  $   BASE STYLE
 //  ----------------------------------------------------------------------------
 .d-table {
+    //  Component CSS Vars
+    //  ------------------------------------------------------------------------
+    --table--fc:           var(--black-600);
+    --table--bc:           var(--black-200);
+    --table--bgc:          var(--white);
```
Comment (ryandialpad) [inline lib/less/components/tables.less]: These vars make creating themed variants really easy 💯

File: lib/less/components/tables.less [hunk 4]
```diff
@@ -13,40 +13,61 @@
...
+
     font-size: var(--fs14);
-    line-height: var(--lh-normal);
+    line-height: var(--lh2); 
+
+    background-color: var(--table--bgc);
+
+    // Table Head Styles
+    thead {
+        border-bottom: var(--su2) solid var(--table--bc);
```
Comment (dialpad-drew) [inline lib/less/components/tables.less:40]: Column headers (`th` in the `thead`) should have be all uppercase and have a 12px font, not 14px.

File: lib/less/components/tables.less (hunk 1 above)
Comment (dialpad-drew) [reply lib/less/components/tables.less]: Didn't realize we had this. I actually want to keep it. Here's what I'm proposing:

For base tables, this is `black-050` and for inverted tables, this is `black-700`. 

Intended UI:

Screen Shot 2021-04-06 at 2 31 08 PM

File: lib/less/components/tables.less (hunk 2 above)
Comment (dialpad-drew) [reply lib/less/components/tables.less]: I think this is something that's being used by the Dialtone site itself. You can see an example in Avatars:

Screen Shot 2021-04-06 at 2 43 51 PM

I'd suggest keeping it, but updating the comment to mention its use by the Dialtone site when `<tbody>` is used multiple times within a single `<table>`.

File: lib/less/components/tables.less [hunk 5]
```diff
@@ -13,40 +13,61 @@
 //  $   BASE STYLE
 //  ----------------------------------------------------------------------------
 .d-table {
+    //  Component CSS Vars
+    //  ------------------------------------------------------------------------
+    --table--fc:           var(--black-600);
```
Comment (dialpad-drew) [inline lib/less/components/tables.less]: The font-colors have slightly more nuance than this.

For base style:
Column and row headers `<th>`: `--black-700`
Non-headers `<td>`: `--black-600`

For inverted style:
Column and row headers `<th>`: `--black-050`
Non-headers `<td>`: `--black-100`

File: lib/less/components/tables.less [hunk 6]
```diff
@@ -13,40 +13,61 @@
...
 
+    // Table Header Styles
+    th {
+        font-weight: var(--fw-bold);
+    }
+
+    // Cell Styles
     th,
     td {
         padding: var(--su8) var(--su4);
```
Comment (dialpad-drew) [inline lib/less/components/tables.less]: Should be `--su16`. I updated a defect in Figma in which the column headers had different paddings. They now all have 16px padding.

File: lib/less/components/tables.less (hunk 4 above)
Comment (ryandialpad) [reply lib/less/components/tables.less:40]: Updated

File: lib/less/components/tables.less (hunk 6 above)
Comment (ryandialpad) [reply lib/less/components/tables.less]: Updated

File: lib/less/components/tables.less (hunk 5 above)
Comment (ryandialpad) [reply lib/less/components/tables.less]: Updated

File: lib/less/components/tables.less (hunk 1 above)
Comment (ryandialpad) [reply lib/less/components/tables.less]: Updated

File: lib/less/components/tables.less (hunk 2 above)
Comment (ryandialpad) [reply lib/less/components/tables.less]: Good call, updated comment

**Review by dialpad-drew — APPROVED** [review]
LGTM!

File: docs/components/tables.html [hunk 7]
```diff
@@ -104,3 +104,107 @@ <h2 class="dialtone-title__section">Inverted Style</h2>
...
+            <td>…</td>
+            <td>…</td>
+            <td>…</td>
+        </tr>
+    </tbody>
+</table>
+{% endhighlight %}
+        </div>
+    </div>
+</section>
```
Comment (ryandialpad) [inline docs/components/tables.html]:
```suggestion
</section>

```

File: lib/less/components/tables.less [hunk 8]
```diff
@@ -55,24 +88,25 @@
...
     }
-}
+
+    &.d-table--inverted {
+        // Row Styles
+        tr:nth-child(even) {
+            background-color: var(--black-700);
+        }
+    }
+}
```
Comment (ryandialpad) [inline lib/less/components/tables.less]:
```suggestion
}

```

**Pull request merged**